Azure/azure-cli
# [Role] Fix #11594: az role assignment create: Only show supported values for --assignee-principal-type

**Description**
Fix #11594: `--assignee-principal-type` documents wrong list of expected values

The REST spec of `RoleAssignmentPropertiesWithScope.principalType` and `RoleAssignmentProperties.principalType` contains invalid values which are used only internally by the service. (https://github.com/Azure/azure-rest-api-specs/issues/11830)

Until the REST spec is fixed, manually override `PrincipalType` from SDK so that only allowed values are shown:

- `User`
- `Group`
- `ServicePrincipal`

**Testing Guide**
See the added tests under `test_role_assignment_create_using_principal_type`.

See email: _BadRequestError: The PrincipalType property 'MSI' is not valid. It must be 'User', 'Group' or 'ServicePrincipal'_
